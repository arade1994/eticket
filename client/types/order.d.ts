import { type Ticket } from "./ticket";

export interface Order {
  id: string;
  userId: string;
  status: OrderStatus;
  expiresAt: Date;
  version: number;
  ticket: Ticket;
}

export declare enum OrderStatus {
  Created = "created",
  Cancelled = "cancelled",
  AwaitingPayment = "awaiting:payment",
  Complete = "complete",
}
