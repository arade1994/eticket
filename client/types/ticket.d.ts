export interface Ticket {
  id: string;
  title: string;
  price: number;
  userId: string;
  category: string;
  version: number;
}
