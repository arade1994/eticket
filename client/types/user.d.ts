import { type IncomingMessage } from "http";

export interface User {
  id: string;
  firstName: string;
  lastName: string;
  age: number;
  email: string;
  password: string;
}

export interface Rating {
  id: string;
  comment: string;
  rate: number;
  userId: string;
  ratedUserId: string;
}

export interface RequestWithUser extends IncomingMessage {
  currentUser?: { id: string; email: string };
}
