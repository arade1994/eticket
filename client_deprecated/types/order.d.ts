import { type Ticket } from "./ticket";

export interface Order {
  id: string;
  userId: string;
  status: OrderStatus;
  expiresAt: string;
  version: number;
  ticket: Ticket;
}

export declare enum OrderStatus {
  Created = "created",
  Cancelled = "cancelled",
  AwaitingPayment = "awaiting:payment",
  Complete = "complete",
}
