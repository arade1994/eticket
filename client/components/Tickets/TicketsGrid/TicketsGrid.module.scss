.ticketsContainer {
  display: flex;
  flex-direction: column;
  gap: 2rem;
}

.ticketsGrid {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  gap: 2rem;
}
