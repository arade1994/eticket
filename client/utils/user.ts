import { Ticket } from "../types/ticket";
import { User } from "../types/user";

export const getNumOfCreatedTickets = (user: User, tickets: Ticket[]) =>
  tickets?.filter((ticket) => ticket.userId === user.id)?.length;
