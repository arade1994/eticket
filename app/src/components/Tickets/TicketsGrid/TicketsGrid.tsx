import { useMemo } from "react";

import { type Ticket } from "../../../types/ticket";
import { type Rating, type User } from "../../../types/user";
import TicketCard from "../TicketCard/TicketCard";
import TicketsPagination from "../TicketsPagination/TicketsPagination";

import classes from "./TicketsGrid.module.scss";

interface Props {
  ratings: Rating[];
  tickets: Ticket[];
  users: User[];
  currentPage: number;
  itemsPerPage: {
    value: number;
    label: string;
  };
  onPageChange: (page: number) => void;
}

const TicketsGrid = ({
  ratings,
  tickets,
  users,
  currentPage,
  itemsPerPage,
  onPageChange,
}: Props) => {
  const totalPages = Math.ceil(tickets.length / itemsPerPage.value);

  const paginatedTickets = useMemo(
    () =>
      tickets.slice(
        (currentPage - 1) * itemsPerPage.value,
        currentPage * itemsPerPage.value
      ),
    [currentPage, tickets, itemsPerPage.value]
  );

  return (
    <div className={classes.ticketsContainer}>
      <div className={classes.ticketsGrid}>
        {paginatedTickets.map((ticket) => {
          const user = users.find((user) => user.id === ticket.userId);
          return (
            <TicketCard
              key={ticket.id}
              ratings={ratings}
              ticket={ticket}
              user={user}
            />
          );
        })}
      </div>
      {totalPages > 1 && (
        <TicketsPagination
          currentPage={currentPage}
          totalPages={totalPages}
          onPageChange={onPageChange}
        />
      )}
    </div>
  );
};

export default TicketsGrid;
