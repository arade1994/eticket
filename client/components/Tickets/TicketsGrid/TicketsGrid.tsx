import React from "react";

import classes from "./TicketsGrid.module.scss";
import { Ticket } from "../../../types/ticket";
import { Rating, User } from "../../../types/user";
import TicketCard from "../TicketCard/TicketCard";

interface Props {
  ratings: Rating[];
  tickets: Ticket[];
  users: User[];
}

const TicketsGrid = ({ ratings, tickets, users }: Props) => {
  return (
    <div className={classes.ticketsGrid}>
      {tickets.map((ticket) => {
        const user = users.find((user) => user.id === ticket.userId);

        return (
          <TicketCard
            key={ticket.id}
            ticket={ticket}
            user={user}
            ratings={ratings}
          />
        );
      })}
    </div>
  );
};

export default TicketsGrid;
