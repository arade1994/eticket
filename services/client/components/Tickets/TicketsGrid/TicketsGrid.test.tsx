import { describe, expect, test } from "vitest";

import { render } from "@testing-library/react";

import mockDb from "../../../mock/db.json";
import { type Ticket } from "../../../types/ticket";
import { type Rating, type User } from "../../../types/user";

import TicketsGrid from "./TicketsGrid";

const mockTickets = mockDb.tickets as unknown as Ticket[];
const mockRatings = mockDb.ratings as unknown as Rating[];
const mockUsers = mockDb.users as unknown as User[];

describe("<TicketsGrid />", () => {
  test("it should render tickets as expected", () => {
    const { baseElement } = render(
      <TicketsGrid
        ratings={mockRatings}
        tickets={mockTickets}
        users={mockUsers}
      />
    );

    expect(baseElement).toMatchSnapshot();
  });
});
