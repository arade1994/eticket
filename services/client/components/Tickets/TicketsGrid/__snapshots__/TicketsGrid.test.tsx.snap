// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<TicketsGrid /> > it should render tickets as expected 1`] = `
<body>
  <div>
    <div
      class="_ticketsContainer_f2584c"
    >
      <div
        class="_ticketsGrid_f2584c"
      >
        <a
          class="_ticketCard_e6793d"
          href="/tickets/t1"
        >
          <img
            alt="Ticket cover photo"
            class="_coverPhoto_e6793d"
            data-nimg="1"
            decoding="async"
            height="200"
            loading="lazy"
            src="/_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=640&q=75"
            srcset="/_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=384&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=640&q=75 2x"
            style="color: transparent;"
            width="300"
          />
          <div
            class="_ticketContent_e6793d"
          >
            <h4
              class="_ticketTitle_e6793d"
            >
              NBA Finals Game 7
            </h4>
            <div
              class="_ticketPrice_e6793d"
            >
              <p>
                $
                450.00
              </p>
            </div>
            <div
              class="_ticketSeller_e6793d"
            >
              <p>
                Seller: 
                Ante Rade
              </p>
            </div>
            <span
              class="_ticketCategory_e6793d"
            >
              sport game
            </span>
          </div>
        </a>
        <a
          class="_ticketCard_e6793d"
          href="/tickets/t2"
        >
          <img
            alt="Ticket cover photo"
            class="_coverPhoto_e6793d"
            data-nimg="1"
            decoding="async"
            height="200"
            loading="lazy"
            src="/_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=640&q=75"
            srcset="/_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=384&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=640&q=75 2x"
            style="color: transparent;"
            width="300"
          />
          <div
            class="_ticketContent_e6793d"
          >
            <h4
              class="_ticketTitle_e6793d"
            >
              Taylor Swift Eras Tour
            </h4>
            <div
              class="_ticketPrice_e6793d"
            >
              <p>
                $
                280.00
              </p>
            </div>
            <div
              class="_ticketSeller_e6793d"
            >
              <p>
                Seller: 
                Marcus Millford
              </p>
              <span
                class="_rating_e6793d"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                  />
                </svg>
                 
                5.0
              </span>
            </div>
            <span
              class="_ticketCategory_e6793d"
            >
              music event
            </span>
          </div>
        </a>
        <a
          class="_ticketCard_e6793d"
          href="/tickets/t3"
        >
          <img
            alt="Ticket cover photo"
            class="_coverPhoto_e6793d"
            data-nimg="1"
            decoding="async"
            height="200"
            loading="lazy"
            src="/_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=640&q=75"
            srcset="/_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=384&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=640&q=75 2x"
            style="color: transparent;"
            width="300"
          />
          <div
            class="_ticketContent_e6793d"
          >
            <h4
              class="_ticketTitle_e6793d"
            >
              Hamilton Musical
            </h4>
            <div
              class="_ticketPrice_e6793d"
            >
              <p>
                $
                180.00
              </p>
            </div>
            <div
              class="_ticketSeller_e6793d"
            >
              <p>
                Seller: 
                Matt Fitzgerald
              </p>
              <span
                class="_rating_e6793d"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                  />
                </svg>
                 
                4.0
              </span>
            </div>
            <span
              class="_ticketCategory_e6793d"
            >
              theatre play
            </span>
          </div>
        </a>
        <a
          class="_ticketCard_e6793d"
          href="/tickets/t4"
        >
          <img
            alt="Ticket cover photo"
            class="_coverPhoto_e6793d"
            data-nimg="1"
            decoding="async"
            height="200"
            loading="lazy"
            src="/_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=640&q=75"
            srcset="/_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=384&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=640&q=75 2x"
            style="color: transparent;"
            width="300"
          />
          <div
            class="_ticketContent_e6793d"
          >
            <h4
              class="_ticketTitle_e6793d"
            >
              Champions League Final
            </h4>
            <div
              class="_ticketPrice_e6793d"
            >
              <p>
                $
                320.00
              </p>
            </div>
            <div
              class="_ticketSeller_e6793d"
            >
              <p>
                Seller: 
                Marin MiletiÄ‡
              </p>
            </div>
            <span
              class="_ticketCategory_e6793d"
            >
              sport game
            </span>
          </div>
        </a>
        <a
          class="_ticketCard_e6793d"
          href="/tickets/t5"
        >
          <img
            alt="Ticket cover photo"
            class="_coverPhoto_e6793d"
            data-nimg="1"
            decoding="async"
            height="200"
            loading="lazy"
            src="/_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=640&q=75"
            srcset="/_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=384&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=640&q=75 2x"
            style="color: transparent;"
            width="300"
          />
          <div
            class="_ticketContent_e6793d"
          >
            <h4
              class="_ticketTitle_e6793d"
            >
              Ed Sheeran Concert
            </h4>
            <div
              class="_ticketPrice_e6793d"
            >
              <p>
                $
                150.00
              </p>
            </div>
            <div
              class="_ticketSeller_e6793d"
            >
              <p>
                Seller: 
                John Washington
              </p>
            </div>
            <span
              class="_ticketCategory_e6793d"
            >
              music event
            </span>
          </div>
        </a>
        <a
          class="_ticketCard_e6793d"
          href="/tickets/t6"
        >
          <img
            alt="Ticket cover photo"
            class="_coverPhoto_e6793d"
            data-nimg="1"
            decoding="async"
            height="200"
            loading="lazy"
            src="/_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=640&q=75"
            srcset="/_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=384&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=640&q=75 2x"
            style="color: transparent;"
            width="300"
          />
          <div
            class="_ticketContent_e6793d"
          >
            <h4
              class="_ticketTitle_e6793d"
            >
              The Lion King
            </h4>
            <div
              class="_ticketPrice_e6793d"
            >
              <p>
                $
                120.00
              </p>
            </div>
            <div
              class="_ticketSeller_e6793d"
            >
              <p>
                Seller: 
                Sarah Johnson
              </p>
            </div>
            <span
              class="_ticketCategory_e6793d"
            >
              theatre play
            </span>
          </div>
        </a>
        <a
          class="_ticketCard_e6793d"
          href="/tickets/t7"
        >
          <img
            alt="Ticket cover photo"
            class="_coverPhoto_e6793d"
            data-nimg="1"
            decoding="async"
            height="200"
            loading="lazy"
            src="/_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=640&q=75"
            srcset="/_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=384&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=640&q=75 2x"
            style="color: transparent;"
            width="300"
          />
          <div
            class="_ticketContent_e6793d"
          >
            <h4
              class="_ticketTitle_e6793d"
            >
              World Cup Match
            </h4>
            <div
              class="_ticketPrice_e6793d"
            >
              <p>
                $
                400.00
              </p>
            </div>
            <div
              class="_ticketSeller_e6793d"
            >
              <p>
                Seller: 
                Michael Chen
              </p>
            </div>
            <span
              class="_ticketCategory_e6793d"
            >
              sport game
            </span>
          </div>
        </a>
        <a
          class="_ticketCard_e6793d"
          href="/tickets/t8"
        >
          <img
            alt="Ticket cover photo"
            class="_coverPhoto_e6793d"
            data-nimg="1"
            decoding="async"
            height="200"
            loading="lazy"
            src="/_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=640&q=75"
            srcset="/_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=384&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fcover.png&w=640&q=75 2x"
            style="color: transparent;"
            width="300"
          />
          <div
            class="_ticketContent_e6793d"
          >
            <h4
              class="_ticketTitle_e6793d"
            >
              Coldplay World Tour
            </h4>
            <div
              class="_ticketPrice_e6793d"
            >
              <p>
                $
                200.00
              </p>
            </div>
            <div
              class="_ticketSeller_e6793d"
            >
              <p>
                Seller: 
                Emma Wilson
              </p>
            </div>
            <span
              class="_ticketCategory_e6793d"
            >
              music event
            </span>
          </div>
        </a>
      </div>
      <div
        class="_pagination_cadbd0"
      >
        <button
          class="_pageButton_cadbd0 _navButton_cadbd0"
          disabled=""
          id="paginate-left"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
            />
          </svg>
        </button>
        <button
          class="_pageButton_cadbd0 _active_cadbd0"
          id="page-1"
        >
          1
        </button>
        <button
          class="_pageButton_cadbd0 "
          id="page-2"
        >
          2
        </button>
        <button
          class="_pageButton_cadbd0 "
          id="page-3"
        >
          3
        </button>
        <button
          class="_pageButton_cadbd0 "
          id="page-4"
        >
          4
        </button>
        <button
          class="_pageButton_cadbd0 _navButton_cadbd0"
          id="paginate-right"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 320 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;
